### PR TITLE
Fix #9 Creation of an Apartment

### LINKED ISSUE #9: Fail fast - Création d'un apartement depuis un XML
https://github.com/av1m/Apartments/blob/f10f6a0785eb74f1e5b3178155ae24b7e714fb98/src/main/java/io/github/oliviercailloux/y2018/apartments/readapartments/ReadApartmentsXMLFormat.java#L59

Afin de respecter le fail fast, il faudrait faire remplacer les opérations ternaires internes aux appels des setters par des appels seulement si le champs existe. 

_Note :
Dans le cas où un champ obligatoire n'est pas renseigné, le builder soulèvera une exception si nécessaire. Dans le cas où la variable n'est pas obligatoire, la classe Apartment s'occupera de mettre une valeur par défaut. Si nous gardons le code comme tel, le changement d'une valeur par défaut impliquerait de la modifier dans la classe Apartment ET dans le readApartmentXMLFormat (pas très implicit)._

### PR DESCRIPTION
**Pull request by GabG02**
During the 1st iteration, we created a duplication of default values. 
By using Apartment Builder class, we can easily use Builder's verification instead of duplicating another logic in XML Reader class.

### CODE REVIEW

File: src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/xml/ReadApartmentsXMLFormat.java [hunk 1]
```diff
@@ -33,63 +33,79 @@ public ReadApartmentsXMLFormat() {
...
+    }
+    if (prop.containsKey("floorArea")) {
+      apartBuilder.setFloorArea(Double.parseDouble(prop.getProperty("floorArea")));
+    }
+    return apartBuilder;
+  }
+
+  public Builder setBooleanValues(Builder apartBuilder) {
+    if (prop.containsKey("terrace")) {
+      apartBuilder.setTerrace(Boolean.valueOf(prop.getProperty("terrace")));
```
Comment (MorganeFt) [inline src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/xml/ReadApartmentsXMLFormat.java]: Il ne serait pas mieux d'utiliser `.parseBoolean()` à la place de `.valueOf()` ?

File: src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/xml/ReadApartmentsXMLFormat.java [hunk 2]
```diff
@@ -33,63 +33,79 @@ public ReadApartmentsXMLFormat() {
    *     values for the other parameters.
```
Comment (MorganeFt) [inline src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/xml/ReadApartmentsXMLFormat.java:33]: Ligne 18 : l'attribut prop pourrait être passé en `final`

File: src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/xml/ReadApartmentsXMLFormat.java (hunk 1 above)
Comment (GabG02) [reply src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/xml/ReadApartmentsXMLFormat.java]: fix in c6be9ff98c1c38f413b86b273dab10185b34439c

File: src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/xml/ReadApartmentsXMLFormat.java (hunk 2 above)
Comment (GabG02) [reply src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/xml/ReadApartmentsXMLFormat.java:33]: fixed 951139fd655d582435da993df6efd8395a3cf4dc

**Review by MorganeFt — APPROVED** [review]

**Review by oliviercailloux — APPROVED** [review]

**Pull request merged**